numismatic/feeds/poloniex.py

```diff
@@ -47,9 +40,49 @@ def handle_trades_and_orders(msg, subscription):
             one message with both information embedded
         '''
 
-        PoloniexWebsocketClient._handle_orderbook(msg, subscription)
-        PoloniexWebsocketClient._handle_trade(msg, subscription)
-        return STOP_HANDLERS
+        # FIXME: Should probably loop through messages
+        # instead of calling them by index
+        if len(msg) == 3:
+            if len(msg[2]) == 1:
+                PoloniexWebsocketClient._handle_order_book(msg, subscription)
+                PoloniexWebsocketClient._handle_orderbook_removemodify(msg, subscription)
+            elif len(msg[2]) == 2:
+                PoloniexWebsocketClient._handle_orderbook_removemodify(msg, subscription)
+                PoloniexWebsocketClient._handle_trade(msg, subscription)
+
+            return STOP_HANDLERS
+
+    @staticmethod
+    def _handle_order_book(msg, subscription):
+        '''
+            Poloniex order book, this will provide a snapshot
+            of the order book
+            TODO Put format here
+        '''
+
+        if msg[2][0][0] == 'i':
```

Comment:
I think keep the other conditions that you had here, just in one compound statement, so

    if len(msg)==3 and len(msg[1])==1 and msg[2][0][0]=='i':

or something like that.

Reply:
This as a whole needs to change. Rather than refer to the messages by index, I should be interating over it's contents. I will do this when I get 10 minutes this evening.